### PR TITLE
TDD

### PR DESCRIPTION
**Pull request by AbbasSarwar**
### Features
* Factorial Method
* FizzBuzz method
* Tests are written before

### CODE REVIEW

**Review by frank1738 — COMMENTED** [review]
Hi Team, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- [ ] You have submitted a project with failing linter checks

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: solver.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      results
+    end
+  end
+
+  def fizz_buzz(num)
+    return 'FizzBuzz' if (num % 3).zero? && (num % 5).zero?
+    return 'Fizz' if (num % 3).zero?
+    return 'Buzz' if (num % 5).zero?
+
+    num
```
Comment (danmainah) [inline solver.rb]: - [ ] @AbbasSarwar and @yegonkimutai great work adding fizz buzz method, however when a number is not divisible by 3 or 5 or both it should return a string of the number
image

File: spec/solver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    it 'return "Buzz" on numbers divisible by 5' do
+      expect(solver.fizz_buzz(25)).to eq('Buzz')
+    end
+
+    it 'return "Fizz" on numbers divisible by 3' do
+      expect(solver.fizz_buzz(15)).to eq('FizzBuzz')
+    end
+
+    it 'return number on numbers not divisible by 3 or 5' do
+      expect(solver.fizz_buzz(11)).to eq(11)
```
Comment (danmainah) [inline spec/solver_spec.rb]: - [ ] @AbbasSarwar and @yegonkimutai great work on your test, however, this test should expect the number to be a string and not an integer.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @AbbasSarwar and @yegonkimutai , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Used TDD to first write tests and then write functions in the solver class.
✔️  Ensured all tests are passing


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Adding how to run and test the project after cloning it


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by amejid — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- Linter checks passing :heavy_check_mark: 
- Changes implemented :heavy_check_mark: 
- Tests passing :heavy_check_mark: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**